pkg/apis/batch/types.go

```diff
@@ -340,6 +352,19 @@ type JobSpec struct {
 	//
 	// +optional
 	Suspend *bool
+
+	// podReplacementPolicy specifies when to create replacement Pods.
+	// Possible values are:
+	// - TerminatingOrFailed means to create a replacement Pod when the previously
+	//   created Pod is terminating or failed.
```

Comment:
should this mention succeeded as well?

Reply:
https://github.com/kubernetes/kubernetes/pull/117015#discussion_r1265658804

We created a API only review for the api-reviewers so questions about that can go there.  It was approved so I think it is sufficient.  

I don't think it should be succeeded as we don't recreate on success.

Reply:
the failed has the same semantic as in `Failed` where we mention it, no?

```
	// - Failed means to wait until a previously created Pod is fully terminated (has phase
	//   Failed or Succeeded) before creating a replacement Pod.
```


It might be good to also make a short explanation about the problem with succeeded counting against failed: https://github.com/kubernetes/enhancements/blob/master/keps/sig-apps/3939-allow-replacement-when-fully-terminated/README.md#the-default-job-controller-behavior

Reply:
ack, moved to the API PR